pkg/ccl/backupccl/file_sst_sink.go

```diff
@@ -294,46 +299,79 @@ func (s *fileSSTSink) write(ctx context.Context, resp exportedSpan) error {
 	return nil
 }
 
-func (s *fileSSTSink) copyPointKeys(ctx context.Context, dataSST []byte) error {
+func isMidRow(resumeKey, maxPointKey, maxRangeEnd roachpb.Key) bool {
+	if len(resumeKey) == 0 {
+		return false
+	}
+	maxKey := maxPointKey
+	if maxKey.Compare(maxRangeEnd) < 0 {
+		maxKey = maxRangeEnd
+	}
+
+	maxRowKey, err := keys.EnsureSafeSplitKey(maxKey)
+	if err != nil {
```

Comment:
it seems like we should return an error if `EnsureSafeSplitKey` returns an error after processing a point key.

Reply:
Isn't true just as good? This doesn't seem worth failing a backup over, when we could just not flush.

What about logging a warning?

Reply:
Could we add a test build assertion? It gives me the heebie geebies that any call to `EnsureSafeSplitKey` could error unexpectedly.

Reply:
Actually, I'm pretty sure if _either_ of these return any error, we _can_ flush since we know we aren't mid-row (as a mid-row key will not return an error)